### PR TITLE
Delete get_ensemble_by_name from `LocalStorage`

### PR DESCRIPTION
**Pull request by dafeda**
Ensemble names are only unique within a single experiment.

## Breaking Change

- Will cause `semeio` to fail, so this PR will need to be merged: https://github.com/equinor/semeio/pull/640
-  Will cause `everest to fail

### CODE REVIEW

**Comment by DanSava** [comment]
Are experiment names unique?

**Comment by dafeda** [comment]
> Are experiment names unique?

Yes they are. One reason why want this is that it makes it easier to read plots since we use `experiment_name:ensemble_name`.

File: src/ert/storage/migration/block_fs.py [hunk 1]
```diff
@@ -314,6 +314,7 @@ def migrate_case(storage: Storage, path: Path, stack: ExitStack) -> None:
         parameters=parameter_configs,
         responses=response_configs,
         observations=ert_config.observations,
+        name="migrate-case",
```
Comment (yngve-sk) [inline src/ert/storage/migration/block_fs.py:317]: Should this be made unique or is it ok for cases to be migrated (from blockfs in this case so maybe no that many) w/ the same experiment name? Logging statement above is
`    logger.info(f"Migrating case '{path.name}'")`

**Comment by DanSava** [comment]
> > Are experiment names unique?
> 
> Yes they are. One reason why want this is that it makes it easier to read plots since we use `experiment_name:ensemble_name`.

I could not find any logic in ERT that enforces the experiment name to be unique, but it might be that I did not look hard enough. 

If the experiment name is unique and this is checked when a new experiment is created then to make Everest work with these changes we would need to update:

*  `ert/src/ert/simulator/batch_simulator.py`

```python
experiment = storage.create_experiment(
    parameters=self.ert_config.ensemble_config.parameter_configuration,
    responses=self.ert_config.ensemble_config.response_configuration,
)
```
to 
```python
experiment = storage.create_experiment(
    parameters=self.ert_config.ensemble_config.parameter_configuration,
    responses=self.ert_config.ensemble_config.response_configuration,
    name=f"experiment_{case_name}"
)
```

**Review by yngve-sk — APPROVED** [review]
Recommend double checking wrt uniqueness of experiment name in some edge cases, otherwise LGTM

**Comment by dafeda** [comment]
> > > Are experiment names unique?
> > 
> > 
> > Yes they are. One reason why want this is that it makes it easier to read plots since we use `experiment_name:ensemble_name`.
> 
> I could not find any logic in ERT that enforces the experiment name to be unique, but it might be that I did not look hard enough.
> 
> If the experiment name is unique and this is checked when a new experiment is created then to make Everest work with these changes we would need to update:
> 
> * `ert/src/ert/simulator/batch_simulator.py`
> 
> ```python
> experiment = storage.create_experiment(
>     parameters=self.ert_config.ensemble_config.parameter_configuration,
>     responses=self.ert_config.ensemble_config.response_configuration,
> )
> ```
> 
> to
> 
> ```python
> experiment = storage.create_experiment(
>     parameters=self.ert_config.ensemble_config.parameter_configuration,
>     responses=self.ert_config.ensemble_config.response_configuration,
>     name=f"experiment_{case_name}"
> )
> ```

You are right I think, the validation is only in the front-end. @larsevj is working on making sure it's not possible to create experiments with non-unique names I think.

File: src/ert/storage/migration/block_fs.py (hunk 1 above)
Comment (dafeda) [reply src/ert/storage/migration/block_fs.py:317]: Valid point, but this not change behavior as the current code does this when no name is specified:

`name = datetime.today().strftime("%Y-%m-%d")`

Will leave as is for now. OK?

**Pull request merged**